firmware/application/recent_entries.hpp

```diff
@@ -151,6 +150,7 @@ class RecentEntriesTable : public Widget {
             const auto item_style = (has_focus() && is_selected_key) ? s.invert() : s;
             draw(entry, target_rect, painter, item_style);
             target_rect += {0, target_rect.height()};
+            set_focusable(true);
```

Comment:
Could you move this outside the loop? on_paint doesn't feel like the right spot for this.
Maybe in the ctor do this? `set_focusable(recent.size() > 0)` or something?

Reply:
Or is the problem that the entries are modified elsewhere so there's no signal to indicate that a new entry has been added. If that's the case, maybe there's also the inverse where you want to set focusable false if the list is cleared.
Maybe `set_focusable(recent.size() > 0)` at the top of paint then?

Reply:
I wasn't sure of the best place to move the "set_focusable()" code so I surely appreciate your advice on that.

Yes, I did consider moving it to the Container code above where entries are added & removed, but it was outside of the RecentEntriesTable class which deals with the widget and outside of the ui namespace.  Hence, I put the code in the paint function where there is already some code dealing with focus.

I agree with you that moving it outside of the loop is more correct and more understandable.  But, I also noticed that the added check for recent.size() or recent.empty() consumes 64 more bytes of code space and I know we're very short on code space.

Nevertheless, I like your recommendation because it has the added advantage that focusable will be set to false if the list is ever cleared, so I have updated the code accordingly.

Reply:
Interesting on the size of that call. That seems surprising. For almost all of the STL containers that’s like a member check so my only guess is that it’s template bloat. :(

We’ll get the size problem sorted so I’m not too concerned.